File: Strategy/Java/src/duck/ModelDuck.java
package eu.svetlin.design.patterns.strategy.duck;

import eu.svetlin.design.patterns.strategy.fly.FlyNoWay;
import eu.svetlin.design.patterns.strategy.quack.Quack;

public class ModelDuck extends Duck{
    public ModelDuck() {
        super();
        flyBehavior = new FlyNoWay();
        quackBehavior = new Quack();
    }

    @Override
    public void display() {
        System.out.println("I'm a model duck");
    }
}

File: Strategy/TypeScript/src/duck/ModelDuck.ts
import Duck from './Duck';
import FlyNoWay from '../fly/FlyNoWay';
import MuteQuack from '../quack/MuteQuack';

class ModelDuck extends Duck {
  constructor() {
    super();
    this.flyBehavior = new FlyNoWay();
    this.quackBehavior = new MuteQuack();
  }

  display() {
    console.log("I'm a model duck");
  }
}

export default ModelDuck;
